extension/firefox/chrome/ShumwayCom.jsm

```diff
@@ -273,6 +291,98 @@ var ShumwayCom = {
           return;
         }
         onSystemResourceCallback = callback;
+      },
+      createLocalConnection: function(connectionName, callback) {
+        if (typeof connectionName !== 'string' ||
+            connectionName[0] !== '_' && connectionName.split(':').length !== 2) {
+          return -1; // LocalConnectionConnectResult.InvalidName
+        }
+        if (localConnectionsRegistry[connectionName]) {
+          return -2; // LocalConnectionConnectResult.AlreadyTaken
+        }
+        var connection = {
+          callback,
+          allowedSecureDomains: Object.create(null),
+          allowedInsecureDomains: Object.create(null)
+        };
+        localConnectionsRegistry[connectionName] = connection;
+        return 0; // LocalConnectionConnectResult.Success
+      },
+      hasLocalConnection: function(connectionName) {
+        return !!_getLocalConnection(connectionName);
+      },
+      closeLocalConnection: function(connectionName) {
+        if (!_getLocalConnection(connectionName)) {
+          return -1; // LocalConnectionCloseResult.NotConnected
+        }
+        delete localConnectionsRegistry[connectionName];
+        return 0; // LocalConnectionCloseResult.Success
+      },
+      sendLocalConnectionMessage: function(connectionName, methodName, argsBuffer, sender,
+                                           senderURL) {
```

Comment:
> We don't current record the URLs of things we load, but that wouldn't be hard to change.

Loading does not mean it has rights to send messages, e.g. swf loads another domain swf that opens a localconnection. Other swfs from first swf's domain shall not be able to send messages to the latter, no?